lib/refinery/i18n/engine.rb

```diff
@@ -27,7 +27,7 @@ def find_or_set_locale
             if ::Refinery::I18n.has_locale?(locale = params[:locale].try(:to_sym))
               ::I18n.locale = locale
             elsif locale.present? && locale != ::Refinery::I18n.default_frontend_locale
-              params[:locale] = ::I18n.locale.to_s = ::Refinery::I18n.default_frontend_locale.to_s
+              params[:locale] = (::I18n.locale = ::Refinery::I18n.default_frontend_locale).to_s
```

Comment:
@bricesanchez I think we should change this to:

```ruby
::I18n.locale = ::Refinery::I18n.default_frontend_locale.to_s
redirect_to(params.merge(locale: ::I18n.locale), notice: "The locale '#{locale}' is not supported.")
```

Reply:
because it seems like it might avoid the parameter assignment error

Reply:
It's what i would like to do first. i will open a new PR.